src/bin/lockin-internal-demo.rs

```diff
@@ -0,0 +1,160 @@
+#![deny(warnings)]
+#![no_std]
+#![no_main]
+#![cfg_attr(feature = "nightly", feature(core_intrinsics))]
+
+// A constant sinusoid to send on the DAC output.
+const DAC_SEQUENCE: [f32; 8] =
+    [0.0, 0.707, 1.0, 0.707, 0.0, -0.707, -1.0, -0.707];
+
+use dsp::{iir_int, lockin::Lockin};
+use hardware::{Adc1Input, Dac0Output, Dac1Output, AFE0, AFE1};
+use stabilizer::hardware;
+
+#[rtic::app(device = stm32h7xx_hal::stm32, peripherals = true, monotonic = rtic::cyccnt::CYCCNT)]
+const APP: () = {
+    struct Resources {
+        afes: (AFE0, AFE1),
+        adc1: Adc1Input,
+        dacs: (Dac0Output, Dac1Output),
+
+        lockin: Lockin,
+    }
+
+    #[init]
+    fn init(c: init::Context) -> init::LateResources {
+        // Configure the microcontroller
+        let (mut stabilizer, _pounder) = hardware::setup(c.core, c.device);
+
+        let lockin = Lockin::new(
+            &iir_int::IIRState::lowpass(1e-3, 0.707, 2.), // TODO: expose
+        );
+
+        // Enable ADC/DAC events
+        stabilizer.adcs.1.start();
+        stabilizer.dacs.0.start();
+        stabilizer.dacs.1.start();
+
+        // Start sampling ADCs.
+        stabilizer.adc_dac_timer.start();
+
+        init::LateResources {
+            lockin,
+            afes: stabilizer.afes,
+            adc1: stabilizer.adcs.1,
+            dacs: stabilizer.dacs,
+        }
+    }
+
+    /// Main DSP processing routine for Stabilizer.
+    ///
+    /// # Note
+    /// Processing time for the DSP application code is bounded by the following constraints:
+    ///
+    /// DSP application code starts after the ADC has generated a batch of samples and must be
+    /// completed by the time the next batch of ADC samples has been acquired (plus the FIFO buffer
+    /// time). If this constraint is not met, firmware will panic due to an ADC input overrun.
+    ///
+    /// The DSP application code must also fill out the next DAC output buffer in time such that the
+    /// DAC can switch to it when it has completed the current buffer. If this constraint is not met
+    /// it's possible that old DAC codes will be generated on the output and the output samples will
+    /// be delayed by 1 batch.
+    ///
+    /// Because the ADC and DAC operate at the same rate, these two constraints actually implement
+    /// the same time bounds, meeting one also means the other is also met.
+    ///
+    /// TODO: Document
+    #[task(binds=DMA1_STR4, resources=[adc1, dacs, lockin], priority=2)]
+    fn process(c: process::Context) {
+        let adc_samples = c.resources.adc1.acquire_buffer();
+        let dac_samples = [
+            c.resources.dacs.0.acquire_buffer(),
+            c.resources.dacs.1.acquire_buffer(),
+        ];
+
+        // DAC0 always generates a fixed sinusoidal output.
+        for (i, value) in DAC_SEQUENCE.iter().enumerate() {
+            // Full-scale gives a +/- 12V amplitude waveform. Scale it down to give +/- 100mV.
```

Comment:
10.24 V should be full scale on both ADC and DAC. Let's say `0.9*i16::MAX` to not clip anything. But why `1/120`?